Lib/test/test_asyncio/test_events.py

```diff
@@ -1799,6 +1799,7 @@ async def test():
             next(it)
 
 
+@support.requires_subprocess()
 class SubprocessTestsMixin:
```

Comment:
Applying the decorator to the mixin class would cause its subclasses to skip all of their tests, not just the subprocess ones. I've got a possible alternative approach on [the corresponding Android branch](https://github.com/python/cpython/commit/460078deda6a9adaa13d4b4874cf4f0f03b717a8#diff-b125df0ba39acde9a49ff4ce6a554310cab3daf258a2e77c219fce5a8b17c146).

Reply:
Agreed that this is possibly over-enthusiastic in what it is skipping; my inclination is to be explicit on each test that should be skipped, rather than programmatically excluding every test, on the off chance that a test is ever added to the mixin that *doesn't* explicitly require subprocess.